<a href="https://colab.research.google.com/github/TanChen168/ExploratoryVizWeek3/blob/main/AverageHeight.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [22]:
import pandas as pd

filename = '/content/sample_data/athleteEventsNoPersonal.csv'

df = pd.read_csv(filename)
df


ID   Age  ...                                     Event Medal
0           5  21.0  ...          Speed Skating Women's 500 metres   NaN
1           5  21.0  ...        Speed Skating Women's 1,000 metres   NaN
2           5  25.0  ...          Speed Skating Women's 500 metres   NaN
3           5  25.0  ...        Speed Skating Women's 1,000 metres   NaN
4           5  27.0  ...          Speed Skating Women's 500 metres   NaN
...       ...   ...  ...                                       ...   ...
40611  135567  28.0  ...    Rowing Men's Lightweight Coxless Fours   NaN
40612  135570  27.0  ...  Ski Jumping Men's Large Hill, Individual   NaN
40613  135570  27.0  ...        Ski Jumping Men's Large Hill, Team   NaN
40614  135571  30.0  ...                      Bobsleigh Men's Four   NaN
40615  135571  34.0  ...                      Bobsleigh Men's Four   NaN

[40616 rows x 12 columns]

1.a How many rows? - 40616


1.b How many columns? - 12


1.c Which columns have missing values? - Age, Height, Medal got missing values. Atheletes may not win medals in all events that's why it has missing values.

In [21]:
df.info()
df.isna().sum()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 40616 entries, 0 to 40615
Data columns (total 12 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   ID      40616 non-null  int64  
 1   Age     39143 non-null  float64
 2   Height  31615 non-null  float64
 3   Team    40616 non-null  object 
 4   NOC     40616 non-null  object 
 5   Games   40616 non-null  object 
 6   Year    40616 non-null  int64  
 7   Season  40616 non-null  object 
 8   City    40616 non-null  object 
 9   Sport   40616 non-null  object 
 10  Event   40616 non-null  object 
 11  Medal   5917 non-null   object 
dtypes: float64(2), int64(2), object(8)
memory usage: 3.7+ MB


ID            0
Age        1473
Height     9001
Team          0
NOC           0
Games         0
Year          0
Season        0
City          0
Sport         0
Event         0
Medal     34699
dtype: int64

1.d. How many entries correspond to the city of London? - 3370 entries

In [30]:
city = df.loc[df['City'] == 'London', :]
len(city)

3370

1.e. What age is the youngest athlete in our sample data? - 11

In [ ]:
age = df['Age'].min()
age

11.0

2.a. How many athletes who participated in the Sport "Tug-Of-War" were from Team "Sweden" in our sample dataset? - 6

In [32]:
City_filter = df['Team'] == 'Sweden'
Event_filter = df['Sport']=='Tug-Of-War'
Sweden_ToF = df.loc[City_filter & Event_filter, :]
len(Sweden_ToF)

6

2.b. How many Gold medals were awarded to athletes who were over 40 years old? - 66 gold medals

In [33]:
gold_filter = df['Medal'] == 'Gold'
yo40_filter = df['Age'] > 40
Gold40yo = df.loc[gold_filter & yo40_filter, :]
len(Gold40yo)

66

3. Determine the average height of:
     a. All rows included - 175.38

In [34]:
round(df['Height'].mean(), 2)

175.38

3. Determine the average height of:
     b. All rows included, but grouped by Event

In [38]:
height = df.groupby(['Event'])['Height'].mean()
height


Event
Alpine Skiing Men's Combined                      178.547170
Alpine Skiing Men's Downhill                      178.211679
Alpine Skiing Men's Giant Slalom                  176.739394
Alpine Skiing Men's Slalom                        176.482143
Alpine Skiing Men's Super G                       179.850575
                                                     ...    
Wrestling Women's Flyweight, Freestyle            158.100000
Wrestling Women's Heavyweight, Freestyle          173.444444
Wrestling Women's Light-Heavyweight, Freestyle    170.000000
Wrestling Women's Lightweight, Freestyle          162.666667
Wrestling Women's Middleweight, Freestyle         164.533333
Name: Height, Length: 726, dtype: float64

3.c. Bonus - Drop duplicate athlete and get the height

In [39]:
df.value_counts('ID')
one_per_athlete = df.drop_duplicates(subset = 'ID', keep = 'first')
one_per_athlete.groupby(['Event'])['Height'].mean()

Event
Alpine Skiing Men's Combined                      178.800000
Alpine Skiing Men's Downhill                      178.177215
Alpine Skiing Men's Giant Slalom                  176.379310
Alpine Skiing Men's Slalom                        175.583333
Alpine Skiing Men's Super G                       178.933333
                                                     ...    
Wrestling Women's Flyweight, Freestyle            158.000000
Wrestling Women's Heavyweight, Freestyle          172.000000
Wrestling Women's Light-Heavyweight, Freestyle    170.000000
Wrestling Women's Lightweight, Freestyle          161.700000
Wrestling Women's Middleweight, Freestyle         165.100000
Name: Height, Length: 609, dtype: float64